addon/index.js

```diff
@@ -186,6 +188,8 @@ function installExperiment(experiment) {
 }
 
 function uninstallSelf() {
+  // HACK: onUnload often doesn't seem to get called, here.
+  exports.onUnload('uninstall');
```

Comment:
Buuuuut... if I just revert what I did to index.js, the page mod changes on detach seem to resolve this immediate issue fine. So, I removed that stuff to just address the retire page thing for this issue